src/picongpu/include/fields/FieldJ.kernel

```diff
@@ -342,7 +342,14 @@ struct KernelAddCurrentToEMF
         const DataSpace<simDim > threadIndex(threadIdx);
         auto fieldJBlock = fieldJ.shift(blockCell);
 
-        ThreadCollective<BlockArea> collective(threadIndex);
+        ThreadCollective<
+            BlockArea,
+            PMacc::math::CT::volume< typename BlockArea::SuperCellSize >::type::value
+        > collective(
+            DataSpaceOperations< BlockArea::Dim >::template map<
+                typename BlockArea::SuperCellSize
+            >( threadIndex )
```

Comment:
I removed the constructor for n-dimensional thread indices. Therefore we need to map the n-dimensional index to a linear index.
In the old code we used a one to one relation between the number of worker and number of elements in a supercell, this is not longer valid for lock step programming